src/analyzers/Infeasibility.jl

```diff
@@ -5,11 +5,57 @@
 
 module Infeasibility
 
-import ModelAnalyzer
 import MathOptInterface as MOI
+import ModelAnalyzer
+
+# This type and the associated function were inspired by IntervalArithmetic.jl
+# Copyright (c) 2014-2021: David P. Sanders & Luis Benet
+struct Interval{T<:Real}
+    lo::T
+    hi::T
+
+    function Interval(lo::T, hi::T) where {T<:Real}
+        @assert lo <= hi
+        return new{T}(lo, hi)
+    end
+end
+
+Base.convert(::Type{Interval{T}}, x::T) where {T<:Real} = Interval(x, x)
+
+Base.zero(::Type{Interval{T}}) where {T<:Real} = Interval(zero(T), zero(T))
+
+Base.iszero(a::Interval) = iszero(a.hi) && iszero(a.lo)
+
+function Base.:+(a::Interval{T}, b::Interval{T}) where {T<:Real}
+    return Interval(a.lo + b.lo, a.hi + b.hi)
+end
+
+function Base.:*(x::T, a::Interval{T}) where {T<:Real}
+    if iszero(a) || iszero(x)
+        return Interval(zero(T), zero(T))
+    elseif x >= zero(T)
+        return Interval(a.lo * x, a.hi * x)
+    end
+    return Interval(a.hi * x, a.lo * x)
+end
+
+function _eval_variables(value_fn::Function, t::MOI.ScalarAffineTerm)
+    return t.coefficient * value_fn(t.variable)
+end
 
-include("intervals.jl")
-include("_eval_variables.jl")
+function _eval_variables(
+    value_fn::Function,
+    f::MOI.ScalarAffineFunction{T},
+) where {T}
+    # TODO: this conversion exists in JuMP, but not in MOI
+    S = Base.promote_op(value_fn, MOI.VariableIndex)
+    U = MOI.MA.promote_operation(*, T, S)
+    out = convert(U, f.constant)
+    for t in f.terms
+        out += _eval_variables(value_fn, t)
+    end
+    return out
```

Comment:
Ditto here: why is this `_eval_variales` needed?

Reply:
because MOI assumes the output type is the same as the ScalarAffineFunction (Float64), but we need the output to be an interval. JuMP does is correctly: https://github.com/jump-dev/ModelAnalyzer.jl/pull/19#discussion_r2078871679